### ISSUE TITLE
IPv6 address can't be specified when attaching a NIC to a VM

### ISSUE THREAD

**Issue by brodriguez-opennebula** [OpenNebula/one#6633]
**Description**

Attaching an IPv6 alias on a VM specifying an address does not work. **This does not happen when the IPv6 is not specified**

**To Reproduce**

- Define an IPv6 range for a Virtual network
```
AR=[
  IP6="2a04:daaa:0:1::2",
  PREFIX_LENGTH="64",
  SIZE="256",
  SLAAC="off",
  TYPE="IP6_STATIC",
]
```

- Try to attach it (as an alias) to a VM
```
$ onevm nic-attach -n 33 -a NIC0 -i 2a04:da00:0:1::fff1 118
[one.vm.attachnic] Cannot get IP/MAC lease from virtual network 33.
```

- The template generated by the command would be the following
```
NIC_ALIAS = [ PARENT = NIC0, NETWORK_ID = 33, IP   = 2a04:da00:0:1::fff1]
```

**Expected behavior**
The specified NIC would get an alias with the specified IPv6 address

**Details**
 - Affected Component: Core
 - Hypervisor: KVM
 - Version: 6.8

**Additional context**
- As commented before, the following command works correctly and attaches an IPv6 alias on the interface
```
$ onevm nic-attach -n 33 -a NIC0 118
```
- IPv6 address ranges can be larger than the amount of MACs that can be assigned to its VMs. That can lead to a source of problems

## Progress Status
- [ ] Code committed
- [ ] Testing - QA
- [ ] Documentation (Release notes - resolved issues, compatibility, known issues)

**Comment by gigatux** [comment]
I've been desperately trying to figure this out, but I'm coming to the conclusion that the _PREFIX_LENGTH_ is being ignored, and _SIZE_ simply refers to the number of individual IPv6 addresses (i.e. /128s), so 2a04:da00:0:1::fff1 is indeed out of the range you have of size 256, which is 2a04:daaa:0:1::2 to 2a04:daaa:0:1::102.

I would love to be corrected, and really hope I'm wrong, but I can't get prefix lengths to work at all with IP6_STATIC address blocks.

**Comment by rsmontero** [comment]
Yes for all networks SIZE is the number of IP addresses from the start. So yes in this case you have

`IP6, IP6+1,....,IP6+SIZE-1`

In case of IP6_STATIC you can define multiple AR's each one with the subset of the full range you are interested in lease to your VMs (i.e. because you have gaps in the addresses space):

```
AR = [ IP6="2a04:daaa:0:1::2", SIZE="512"..."...
AR = [ IP6="2a04:daaa:0:1::fff1", SIZE="512"..."..
....

```
Note that PREFIX_LENGTH is for VM configuration not for computing the address space.

So I think it is working as expected

**Comment by gigatux** [comment]
@rsmontero But this doesn't work in many use cases, e.g. allocating a full /64 to each customer, which is quite normal.

A /64 has 2^64 IP addresses (128 - 64 = 64). And 2^64 is obviously the largest number that can fit in an unsigned 64-bit integer, and I'm pretty sure that _SIZE_ ultimately maps to a _long_ so when you try to enter a number bigger than this, it complains that it's not a number.

So, I think there still is a bug here for this common use case, and using any sort of _SIZE_ (unless it uses PREFIX_LENGTH in combination, which is what @brodriguez-opennebula assumed) just isn't fit for purpose.
